sentinel/transitions/index.js

```diff
@@ -324,15 +305,31 @@ const finalize = ({ change, results }, callback) => {
 const applyTransition = ({ key, change, transition }, callback) => {
 
   const _setResult = ok => {
-    const doc = change.doc;
-    if (!doc.transitions) {
-      doc.transitions = {};
-    }
-    doc.transitions[key] = {
-      last_rev: parseInt(change.doc._rev) + 1,
-      seq: change.seq,
-      ok: ok
-    };
+    return new Promise((resolve, reject) => {
+      const info = change.info;
+      info._id = info._id || `${change.id}-info`;
+
+      dbPouch.sentinel.get(info._id)
+      .then(doc => { return doc; })
+      .catch(err => {
+        if(err.status === 404) { return info; }
+        return reject(err);
+      })
+      .then(doc => {
+        doc.transitions = doc.transitions || {};
+        doc.transitions[key] = {
+          last_rev: change.doc._rev,
+          seq: change.seq,
+          ok: ok
+        };
+        dbPouch.sentinel.put(doc, err => {
+          if(err) {
+            logger.error('Error updating metaData', err);
+          }
+          return resolve();
+        });
+      });
+    });
```

Comment:
Once again you've wrapped promise code in a `new Promise`. This is slow and hard to read.